File: Frontend/src/interfaces/v12/invoice/invoiceCreateUpdateDto.ts
import { guid } from "types/guid";
import { InvoiceItemCreateUpdateDto } from "./invoiceItemCreateUpdateDto";

export interface InvoiceCreateUpdateDto {
  customerId?: guid | null;
  customerName?: string | null;
  dateOfSale: string;
  salesPersonId: string;
  warehouseId: guid;
  salesTypeId?: guid | null;
  paymentTypeId?: guid | null;
  paymentReference?: string | null;
  remark?: string | null;
  invoiceItems: InvoiceItemCreateUpdateDto[];
}

File: Wms.Api/Dto/Invoice/InvoiceCreateUpdate/InvoiceCreateUpdateDto.cs
using System.ComponentModel.DataAnnotations;

namespace Wms.Api.Dto.Invoice.InvoiceCreateUpdate
{
    public class InvoiceCreateUpdateDto
    {
        public Guid? CustomerId { get; set; }

        [MaxLength(256)]
        public string? CustomerName { get; set; }

        [Required]
        public DateTime DateOfSale { get; set; }

        [Required]
        [MaxLength(450)]
        public string SalesPersonId { get; set; } = default!;

        [MaxLength(128)]
        public string? EOrderNumber { get; set; }

        public Guid? SalesTypeId { get; set; }

        public Guid? PaymentTypeId { get; set; }

        [MaxLength(128)]
        public string? PaymentReference { get; set; }

        [Required]
        public Guid WarehouseId { get; set; }

        [Required]
        public ICollection<InvoiceItemCreateUpdateDto> InvoiceItems { get; set; } = new List<InvoiceItemCreateUpdateDto>();
    }
}
